src/Compiler/Utilities/range.fs

```diff
@@ -581,3 +581,83 @@ module Range =
                     | None -> mkRange file (mkPos 1 0) (mkPos 1 80)
         with _ ->
             mkRange file (mkPos 1 0) (mkPos 1 80)
+
+module internal FileContent =
```

Comment:
I feel uneasy with this module.
- It adds a global resource for a rather local feature.
- It creates another dependency on the file system.
- It re-reads sources from the file system that have been read before.

Getting the source text out of the checking environment as proposed by @brianrourkeboll would be much cleaner.
A (probably too ambitious) alternative would be to make this a central place for source access, to give ISourceText the role that it was meant to have and to remove file system access elsewhere.